src/client/Microsoft.Identity.Client/ApiConfig/SystemWebViewOptions.cs

```diff
@@ -1,169 +1,176 @@
-            else
-            {
-                throw new PlatformNotSupportedException(RuntimeInformation.OSDescription);
-            }
-#endif
-
-        }
-#endif
-    }
-}
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Logger;
+using Microsoft.Identity.Client.PlatformsCommon.Factories;
+
+namespace Microsoft.Identity.Client
+{
+    // Default browser WebUI is not available on mobile (Android, iOS, UWP), but allow it at runtime
+    // to avoid MissingMethodException
+
+    /// <summary>
+    /// Options for using the default OS browser as a separate process to handle interactive auth.
+    /// MSAL will be listening for the OS browser to finish authenticating, but it cannot close the browser.
+    /// It can however respond with a 200 OK message or a 302 Redirect, which can be configured here.
+    /// For more details see https://aka.ms/msal-net-os-browser
+    /// </summary>
+#if !SUPPORTS_OS_SYSTEM_BROWSER
+    [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+#endif
+    public class SystemWebViewOptions
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public SystemWebViewOptions()
+        {
+            ValidatePlatformAvailability();
+        }
+
+        /// <summary>
+        /// When the user finishes authenticating, MSAL will respond with a 200 OK message,
+        /// which the browser will show to the user. 
+        /// </summary>
+        public string HtmlMessageSuccess { get; set; }
+
+        /// <summary>
+        /// When the user finishes authenticating, but an error occurred,
+        /// MSAL will respond with a 200 OK message, which the browser will show to the user.
+        /// You can use a string format e.g. "An error has occurred: {0} details: {1}"
+        /// </summary>
+        public string HtmlMessageError { get; set; }
+
+        /// <summary>
+        /// When the user finishes authenticating, MSAL will redirect the browser to the given Uri
+        /// </summary>
+        /// <remarks>Takes precedence over <see cref="HtmlMessageSuccess"/></remarks>
+        public Uri BrowserRedirectSuccess { get; set; }
+
+        /// <summary>
+        /// When the user finishes authenticating, but an error occurred, MSAL will redirect the browser to the given Uri
+        /// </summary>
+        /// <remarks>Takes precedence over <see cref="HtmlMessageError"/></remarks>
+        public Uri BrowserRedirectError { get; set; }
+
+        /// <summary>
+        /// This hides privacy prompt displayed on the iOS Devices (ver 13.0+) when set to true.
+        /// By default, it is false and displays the prompt.
+        /// </summary>
+        public bool HidePrivacyPrompt { get; set; } = false;

```

Comment:
For naming purposes ... I would name it `iOSHidePrivacyPrompt` to make it clear it only affects iOS. And the comments should mention prefersEphemeralWebBrowserSession and the fact that the browser will not remember previous logged-in users.

Reply:
Good suggestion. Done.